bundles/io/org.eclipse.smarthome.io.http.auth/src/main/java/org/eclipse/smarthome/io/http/auth/internal/AuthenticationHandler.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.http.auth.internal;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.smarthome.core.auth.Authentication;
+import org.eclipse.smarthome.core.auth.AuthenticationException;
+import org.eclipse.smarthome.core.auth.AuthenticationManager;
+import org.eclipse.smarthome.core.auth.Credentials;
+import org.eclipse.smarthome.io.http.Handler;
+import org.eclipse.smarthome.io.http.HandlerContext;
+import org.eclipse.smarthome.io.http.HandlerPriorities;
+import org.eclipse.smarthome.io.http.auth.CredentialsExtractor;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Request handler which allows to verify authentication.
+ *
+ * @author Łukasz Dywicki - initial contribution.
+ */
+@Component(configurationPid = "org.eclipse.smarthome.io.http.auth")
+public class AuthenticationHandler implements Handler {
+
+    private static final String AUTHENTICATION_ENABLED = "authentication.enabled";
+    private static final String AUTHENTICATION_ENDPOINT = "authentication.loginUri";
+
+    private final Logger logger = LoggerFactory.getLogger(AuthenticationManager.class);
+
+    private final List<CredentialsExtractor<HttpServletRequest>> extractors = new CopyOnWriteArrayList<>();
+
+    private AuthenticationManager authenticationManager;
+
+    // configuration properties
+    private boolean enabled = true;
+    private String loginUri = "/login";
```

Comment:
Use a url that works in ESH out of the box without requiring any configuration - e.g. "/login/index.html", if this is where we'd register the form.